### PR TITLE
Tweak & Fix Promethean Stuff

### PR DESCRIPTION
**Pull request by Mechoid**
Local Spider Screams At Mirror For Making Fool's Mistake of Committing to Local Master.

Tweaks
- Severe starvation reduces Promethean passive regen to half effectiveness. 0.5 - > 0.25
- Severe starvation doubles standard agony from passively regenerating, with an increased (90% instead of 70%) pain limit. 10 more halloss is required to down when undergoing extreme or constant regeneration.
- Prometheans with unsafe body temperatures are unable to passively regenerate, due to thermal equilibrium failure.
- Damage from being splashed with water is tentatively increased due to the speed at which negative stacks dissipate.
- Prometheans must stay still for 60 seconds before passive regeneration will activate.
- Siemens Coefficient properly set so that tasers actually do their proper amount of damage, 4 standard shots on unarmored, 3 with a stun revolver or xenotaser, 2 with a True Xenotaser.

Fixes (From Tuning PR)
- Promethean regen actually works, as Tuning PR caused the datum's heal_rate to become infinitesimal.

### CODE REVIEW

**Comment by Atermonera** [comment]
You do know you can push to a different branch on the remote, or make a new branch on your local from master with the commit, push the new branch and make a PR and everything, and then in either case reset your local master to before the commit, right?

**Comment by Mechoid** [comment]
I do now.

File: code/modules/mob/living/carbon/human/species/station/prometheans.dm [hunk 1]
```diff
@@ -149,10 +150,14 @@ var/datum/species/shapeshifter/promethean/prometheans
 			H.gib()
 
 /datum/species/shapeshifter/promethean/handle_environment_special(var/mob/living/carbon/human/H)
-	var/healing = TRUE	// Switches to FALSE if the environment's bad
-
-	if(H.fire_stacks < 0)	// If you're soaked, you're melting
-		H.adjustToxLoss(2*heal_rate)	// Doubled because 0.5 is miniscule, and fire_stacks are capped in both directions
+	var/healing = TRUE	// Switches to FALSE if healing is not possible at all.
+	var/regen_brute = TRUE
```
Comment (Anewbe) [inline code/modules/mob/living/carbon/human/species/station/prometheans.dm]: Why do you insist on keeping these? Keep things simple.

File: code/modules/mob/living/carbon/human/species/station/prometheans.dm [hunk 2]
```diff
@@ -168,48 +173,53 @@ var/datum/species/shapeshifter/promethean/prometheans
...
+	if(H.bodytemperature > heat_level_1 || H.bodytemperature < cold_level_1)	// If you're too hot or cold, you can't heal.
 		healing = FALSE
 
 	// Heal remaining damage.
 	if(healing)
 		if(H.getBruteLoss() || H.getFireLoss() || H.getOxyLoss() || H.getToxLoss())
-
 			var/nutrition_cost = 0		// The total amount of nutrition drained every tick, when healing
-			var/starve_mod = 1			// Lowers healing, increases agony
+			var/nutrition_debt = 0
```
Comment (Anewbe) [inline code/modules/mob/living/carbon/human/species/station/prometheans.dm]: Explain what this does in a comment, at least?

File: code/modules/mob/living/carbon/human/species/station/prometheans.dm (hunk 1 above)
Comment (Mechoid) [reply code/modules/mob/living/carbon/human/species/station/prometheans.dm]: All four are present for any present (Low pressure causing physical injuries to stop regenerating) or possible future need to toggle a single type of healing, based on a new chemical effect or modifier or what-have-you, without disabling them all.
It's also there to keep the lower checks ordered so it's easier for me to transfer them into proper organ processes later.

File: code/modules/mob/living/carbon/human/species/station/prometheans.dm [hunk 3]
```diff
@@ -187,12 +183,10 @@ var/datum/species/shapeshifter/promethean/prometheans
 		healing = FALSE
 
 	// Heal remaining damage.
-	if(H.fire_stacks >= 0)
-		if(H.getBruteLoss() || H.getFireLoss() || H.getOxyLoss() || H.getToxLoss())
+	if(healing)		if(H.getBruteLoss() || H.getFireLoss() || H.getOxyLoss() || H.getToxLoss())
```
Comment (Atermonera) [inline code/modules/mob/living/carbon/human/species/station/prometheans.dm]: missing newline between these if statements

File: code/modules/mob/living/carbon/human/species/station/prometheans.dm [hunk 4]
```diff
@@ -171,14 +169,12 @@ var/datum/species/shapeshifter/promethean/prometheans
...
-	T = get_turf(H) // Swap over to an actual turf, because we need to get the pressure.
-	if(istype(T)) // Make sure it exists, and is a turf again.
 		var/datum/gas_mixture/environment = T.return_air()
 		var/pressure = environment.return_pressure()
 		var/affecting_pressure = H.calculate_affecting_pressure(pressure)
-		if(affecting_pressure <= hazard_low_pressure)	// If you're in a vacuum, you don't heal physical wounds.
-			regen_brute = FALSE
-			regen_burn = FALSE
+		if(affecting_pressure <= hazard_low_pressure) // Dangerous low pressure stops the regeneration of physical wounds. Body is focusing on keeping them intact rather than sealing.
+			regen_brute = 0
```
Comment (Atermonera) [inline code/modules/mob/living/carbon/human/species/station/prometheans.dm]: These should be FALSE, since it's a toggle

**Pull request merged**